src/perl/lib/wtsi_clarity/util/label.pm

```diff
@@ -83,7 +88,8 @@ sub _format_label {
 
   if ($type eq 'plate') {
 
-    $label = {
+    if ($config->barcode_mint->{"internal_generation"}) {
```

Comment:
It is OK for now, but when we will refactor the whole thing, then we should probably use strategy pattern here to make it more maintainable.